add(remove_tooltip_for_help_icon)[OASIS-3278]

circAssimilate opened the pull request:
**Summary:**
As discussed in https://optimizely.slack.com/archives/C03DQJA7D/p1531418787000064, this will remove an unwanted tooltip from the svg when the "Help" Icon is used. This was being caused by the title element from the svg.

**Screenshot (before tooltip removed):**
![image](https://user-images.githubusercontent.com/7128855/42651596-b2d31620-85c4-11e8-9c64-4b829cabdb6f.png)

jessicalc commented:
I'm on the fence about this. What if we have an icon that someone doesn't know how to read?

I think we need to:
1) Decide on whether HelpPopovers should be triggered by **click** or **hover**, as discussed in https://optimizely.atlassian.net/browse/OASIS-3278. Up until the new Full Stack Environments Dashboard, **all** HelpPopovers were triggered on **click**. I don't think it's a good idea to have them be one or the other. I believe Ron punted this question to @daverau-optimizely in the JIRA ticket. 
2) If we decide that HelpPopovers should be triggerable on hover, we should probably make this configurable rather than ripping out the title outright... see next point!! 
3) We should weigh the risks of removing the titles, especially if the icon's meaning is ambiguous. We use icons not just in the context of HelpPopovers!!! And the titles are actually useful sometimes! 

![image](https://user-images.githubusercontent.com/729524/42652352-00d9696c-85c7-11e8-8f34-ebe4ef5d0cf3.png)

If Dave thinks this is an OK approach, I am fine with this, but I just wanted to surface these concerns now!

circAssimilate commented:
Great points, @jessicalc. I also agree we should keep our popover icons consistent, and now agree that click is better so we can still have these tooltips that may be helpful for tricky icons. 

If @daverau-optimizely agrees, I can abandon and then update to behavior="click" in the app. If we decide that we should do hover for all these poptips (which on the other hand I think is more intuitive for a help popover), then perhaps we should add an `Icon` prop that allows for optional rendering of this `title`.

daverau-optimizely commented:
Sorry @circAssimilate I should have been more clear about being in favor of hiding the title for help icons only, while keeping OUI icon titles intact so any cryptic icons still have clear and helpful titles.

Great points and discussion here @jessicalc, I appreciate and share your perspective on keeping titles intact for most icons.

The reason we have a choice between hover and click is for poptips that contain links, which sometimes is difficult to target with a mouse via pure hover. In our implementation, if you hover on the icon and move your mouse over the content popup (outside of the icon hit area) the popup disappears. Neha and I discussed making our hover targets larger and inclusive of the popup so it's easy to move your mouse cursor from the icon to the content and have the whole popup remain open. This would remove the click to show/hide functionality. We'd also have to consider some accessibility issues around hover vs click.

My thoughts:
1. I'd like to see us remove click and default to hover functionality but with some CSS/JS modifications so the user experience holds up to keyboard navigation. I would love to work with someone on improving poptips. Iñaki and I made some progress together but didn't get a chance to implement all the things we had planned.
2. I like the idea of a prop to hide title, maybe even if `icon == help` we exclude `title` as a default.

circAssimilate commented:
Thanks for the feedback @daverau-optimizely and @jessicalc - definitely see your points and now agree. I have updated this component (and added a test) to exclude the title only when "help" is used, which should suffice for [OASIS-3278](https://optimizely.atlassian.net/browse/OASIS-3278). Let me know what you think!

circAssimilate commented:
Gonna merge this and the next OUI bump should catch it.

jessicalc commented:
YAY! This is  agreat solution.
